### ISSUE TITLE
[data] dyn_bsz to control training data strategy

### ISSUE THREAD

**Issue by Coach257** [ByteDance-Seed/VeOmni#436]
### Feature request

Only use `dync_bsz` to control data strategy, delete useless arguments like `use_rmpad`, `use_rmpad_with_position_ids`. 

### Motivation

Two training strategies:
- Pretraining: Train on desired number of tokens, fullfill the data sequence to `max_seqlen`.
- Finetuning: Train on desired number of samples, pack `batch_size` number of samples.

In origin implementation, we use padding: a batch of two samples with len(sample_a)=2 and len(sample_b)=4: 

torch.tensor([a0,a1, 0,0], [b0,b1,b2,b3])

Actually, we can pack two samples to torch.tensor([a0,a1, b0,b1,b2,b3]). So we can actually unify all the data to `bsz=1` and only need `dyn_bsz` to controll different data strategies. And we don't need ci tests for `padded_bsh` when `bs=1` and `bs>1`.

### Your contribution

Clean arguments & ci & collate_fn

**Comment by Luosuu** [comment]
@LiuzcEECS is looking into this as we need to improve arguments on dynamic batching

**Comment by Coach257** [comment]
Done in this pr #458
